Sources/HDHTTPServer/HDHTTPServer.swift

```diff
@@ -91,6 +91,11 @@ public class HDHTTPServer<SocketHandlerManager: ClientSocketHandlerManager> {
                 acceptSemaphore.wait()
                 acceptQueue.async { [weak handler] in
                     handler?.handle(socket: clientSocket)
+                    // TODO: Close handler if it is ready to be closed, such as:
+                    //
+                    // if handler?.isOpen && handler.isClosing {
+                    //     handler?.close()
+                    // }
```

Comment:
I am not sure that closing a socket in this place is reasonable, or not 🤔 
And also this code of the comment didn't work...  This snippets is not correct, but I have no idea because of my poor intelligence..

Reply:
You can write like below 👍 

```swift
if let isOpen = handler?.isOpen, let isClosing = handler?.isClosing, isOpen && isClosing {
     handler?.close()
}
```

`if let ~` is **Optional binding** that is used in order to unwrap optional.